### ISSUE TITLE
X-Opaque-Id header not propagated when using SearchGuard

### ISSUE THREAD

**Issue by mcouthon** [floragunncom/search-guard#669]
Hi, I have opened this [issue](https://forum.search-guard.com/t/x-opaque-id-header-not-propagated-when-using-searchguard/1422) on the SG forum with the detailed information, configurations and steps to reproduce, so I'm just going to cut to the chase.

We believe that SG is filtering out the `X-Opaque-Id` header that ES uses to identify tasks in the slow logs (and elsewhere). 

[This place](https://github.com/floragunncom/search-guard/blob/master/src/main/java/com/floragunn/searchguard/transport/SearchGuardInterceptor.java#L120) in the code looks like a good suspect for where the filtering is happening.

We're using **SG 24.1** with **ES 6.6.1**.

### Steps to reproduce
Basically run this:
```
curl -s -H "X-Opaque-Id:TEST" http://localhost:9200/_tasks?group_by=parents
```
 once with SG disabled, and once when it is enabled (obviously with `https` instead), and you will see the difference in the child tasks: without SG, all child tasks will look like this:

```
{
              "node": "T4S6DTEoTcCgcM9LmnpfeA",
              "id": 5058,
              "type": "netty",
              "action": "cluster:monitor/tasks/lists[n]",
              "start_time_in_millis": 1554367389883,
              "running_time_in_nanos": 44156,
              "cancellable": false,
              "parent_task_id": "yWmK4i_XRs6WcrPRRUJPyg:790",
              "headers": {
                "X-Opaque-Id": "TEST"
              }
            }
```

with SG enabled, the tasks will look like this:

```
{
          "node": "0MMcRoB8SsKObQaoPgepuA",
          "id": 645,
          "type": "netty",
          "action": "cluster:monitor/tasks/lists[n]",
          "start_time_in_millis": 1554367832918,
          "running_time_in_nanos": 512547,
          "cancellable": false,
          "parent_task_id": "yWmK4i_XRs6WcrPRRUJPyg:348",
          "headers": {}
        }
```

The difference is the lack of the headers in the child task.

**Comment by mcouthon** [comment]
Thanks a lot for the quick fix, @floragunncom. Is there any ETA for a release incorporating it?

**Comment by floragunncom** [comment]
Fixed with v25.0 which was released today
